Pinta/MacInterop/ApplicationEvents.cs

```diff
@@ -43,18 +43,25 @@ public static class ApplicationEvents
 	public static event EventHandler<ApplicationQuitEventArgs> Quit {
 		add {
 			lock (lock_obj) {
+
 				quit += value;
-				if (quitHandlerRef == IntPtr.Zero)
-					quitHandlerRef = Carbon.InstallApplicationEventHandler (quit_delegate, CarbonEventApple.QuitApplication);
+
+				if (quitHandlerRef != IntPtr.Zero)
```

Comment:
For this file I find the original code is easier to understand since it's just a one-liner initialization when the pointer is null

Reply:
@cameronwhite no problem, I just made a commit where I follow your suggestions